src/workerd/api/actor-destroy.c++

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "actor-destroy.h"
+#include <workerd/api/global-scope.h>
+#include <workerd/jsg/ser.h>
+
+namespace workerd::api {
+
+ActorDestroyEvent::ActorDestroyEvent() : ExtendableEvent("actorDestroy"){};
+
+kj::Promise<WorkerInterface::CustomEvent::Result>
+ActorDestroyCustomEventImpl::run(kj::Own<IoContext::IncomingRequest> incomingRequest,
+                                 kj::Maybe<kj::StringPtr> entrypointName) {
+
+  auto& context = incomingRequest->getContext();
+
+  // Mark the request as delivered because we're about to run some JS.
+  incomingRequest->delivered();
+
+  EventOutcome outcome = EventOutcome::OK;
+
+  try {
+    co_return co_await context.run(
+        [entrypointName = entrypointName, &context](Worker::Lock& lock) mutable {
+          return lock.getGlobalScope().actorDestroy(
```

Comment:
I think for worker entrypoints we want to have an instance of `jsg::AsyncContextFrame::StorageScope` in the scope where we run javascript, to ensure that the context runs with an async scope set up for tracing.  I think that would be a matter of adding a line to the context.run() lambda, like:
```
    jsg::AsyncContextFrame::StorageScope traceScope = context.makeAsyncTraceScope(lock);
```

I notice other custom event handlers add the `context.run()` promise to `context.waitUntil()`, and then do an await on `incomingRequest->finishScheduled()`, but I assume we're not doing that for actorDestroy() because it only runs for Durable Objects, which have a different interpretation of `context.waitUntil()`?

Reply:
Ah, just noticed Kenton mentioned this is a stopgap for a specific situation, so probably don't need to worry about tracing or what waitUntil does.